decompiler/backend/variabledeclarations.py

```diff
@@ -44,7 +44,10 @@ def visit_assignment(self, instruction: Assignment):
     def visit_loop_node(self, node: LoopNode):
         """Visit the given loop node, taking node of the loop declaration."""
         if isinstance(node, ForLoopNode) and isinstance(node.declaration, Assignment):
-            self._variables.add(node.declaration.destination)
+            if isinstance(node.declaration.destination, Operation):
```

Comment:
Please add a test for variable declarations.

Reply:
Done (just check that there is a for loop with a function (ListOP) in output)